lib/rules/no-else-return.js

```diff
@@ -33,7 +35,45 @@ module.exports = {
          * @returns {void}
          */
         function displayReport(node) {
-            context.report({ node, message: "Unnecessary 'else' after 'return'." });
+            context.report({
+                node,
+                message: "Unnecessary 'else' after 'return'.",
+                fix: fixer => {
+                    const sourceCode = context.getSourceCode();
+                    const startToken = sourceCode.getTokenByRangeStart(node.start);
+                    const elseToken = sourceCode.getTokenBefore(startToken);
+                    const source = sourceCode.getText(node);
+                    const lastIfToken = sourceCode.getTokenBefore(elseToken);
+                    let fixedSource;
+
+                    if (startToken.type === "Punctuator" && startToken.value === "{") {
+                        const firstTokenOfElseBlock = sourceCode.getTokenAfter(startToken);
+
+                        if (lastIfToken.value !== "}" &&
+                                lastIfToken.value !== ";" &&
+                                /^[([/+`-]/.test(firstTokenOfElseBlock.value)) {
+                            return null;
+                        }
+
+                        const endToken = sourceCode.getLastToken(node);
+                        const lastTokenOfElseBlock = sourceCode.getTokenBefore(endToken);
+
+                        if (lastTokenOfElseBlock.value !== ";") {
+                            const nextToken = sourceCode.getTokenAfter(endToken);
+
+                            if (/^[([/+`-]/.test(nextToken.value) ||
```

Comment:
Easy enough to add a check, but Is that actually possible? Since `return` must always be inside a function, so there will always be at least one more token: `}` closing the function